### ISSUE TITLE
Infere prepare data and train flags

### ISSUE THREAD

**Issue by papeye** [papeye/speaker-identification#51]
Now we have:
 PREPARE_TRAIN_DATA = True
 TRAIN = True
which have to be set by hand.
We'd like automatic detection when preparing data nad training is needed. 

PREPARE_TRAIN_DATA should True <=> train_data changed as comparing to previous run. 

For this You can save to a file training audio_paths after preparing the data. Than read it and compare to current run and set the flag accordingly

TRAIN should be True if PREPARE_TRAIN_DATA == True OR hyperparameters (batch_size, epochs, valid_split, scale, shuffle_seed) has changed as comperad to previous run.

Approch could be same as before but save hyperparmeters to a file
